PowerValidatedSolutions.psm1

```diff
@@ -20714,6 +20721,384 @@ Function Get-PasswordPolicyConfig {
+                        "passwordLifetime"
+                        {
+                            $checkReturn = checkRange -name "$($product.Name):$($section.Name):passwordLifetime" -value $customConfig.($product.Name).($section.Name)."passwordLifetime" -minRange 1 -maxRange 99999 -required $true
+                            if (-Not $checkReturn) {
+                                $encounterError = "True"
+                            }
+                        }
+                        "passwordReminder"
+                        {
+                            $checkReturn = checkRange -name "$($product.Name):$($section.Name):passwordReminder" -value $customConfig.($product.Name).($section.Name)."passwordReminder" -minRange 1 -maxRange 99999 -required $true
+                            if (-Not $checkReturn) {
+                                $encounterError = "True"
+                            }
+                        }
+                        "passwordReminderFrequency"
+                        {
+                            $checkReturn = checkRange -name "$($product.Name):$($section.Name):passwordReminderFrequency" -value $customConfig.($product.Name).($section.Name)."passwordReminderFrequency" -minRange 1 -maxRange 99999 -required $true
+                            if (-Not $checkReturn) {
+                                $encounterError = "True"
+                            }
+                        }
+                        "temporaryPassword"
+                        {
+                            $checkReturn = checkRange -name "$($product.Name):$($section.Name):temporaryPassword" -value $customConfig.($product.Name).($section.Name)."temporaryPassword" -minRange 1 -maxRange 99999 -required $true
+                            if (-Not $checkReturn) {
+                                $encounterError = "True"
+                            }
+                        }
+                        # Password Complexity section
+                        "policy"
+                        {	
+							$policyString = $customConfig.($product.Name).($section.Name)."policy"
+                            $customConfig.($product.Name).($section.Name)."policy" | Select-String -Pattern "^retry=(\d+)\s+min=(.+),(.+),(.+),(.+),(.+)" | Foreach-Object {$PasswdPolicyRetryValue, $PasswdPolicyMinValue1, $PasswdPolicyMinValue2, $PasswdPolicyMinValue3, $PasswdPolicyMinValue4, $PasswdPolicyMinValue5 = $_.Matches[0].Groups[1..6].Value}
+                            if ($PasswdPolicyRetryValue -eq "" -or $PasswdPolicyMinValue1 -eq "" -or $PasswdPolicyMinValue2 -eq "" -or $PasswdPolicyMinValue3 -eq "" -or $PasswdPolicyMinValue4 -eq "" -or $PasswdPolicyMinValue5 -eq "") {
+                                Write-LogMessage -Type ERROR -Message "The recommended policy setup should be retry=3 min=disabled,disabled,disabled,disbled,15"  -Colour Red
+								Write-LogMessage -Type ERROR -Message "The custom policy file shows $policyString"  -Colour Red
```

Comment:
Change to:

Write-LogMessage -Type ERROR -Message "Password Policy Configuration File Defined as ($policyString)" -Colour Red

Reply:
resolved in 2948bc273bae34312514db8aba5d7cb584efd8d4